NGitLab/IGroupsClient.cs

```diff
@@ -9,53 +9,183 @@ namespace NGitLab;
 public interface IGroupsClient
 {
     /// <summary>
-    /// Get a list of projects accessible by the authenticated user.
+    /// Get a list of projects accessible to the authenticated user.
+    /// See https://docs.gitlab.com/ee/api/groups.html#list-groups.
     /// </summary>
     IEnumerable<Group> Accessible { get; }
 
+    /// <inheritdoc cref="SearchAsync"/>
     IEnumerable<Group> Search(string search);
 
+    /// <inheritdoc cref="GetAsync"/>
+    /// <param name="search">The search criteria.</param>
     GitLabCollectionResponse<Group> SearchAsync(string search);
 
-    /// <summary>
-    /// Get a list of GitLab groups.
-    /// </summary>
+    /// <inheritdoc cref="GetAsync"/>
     IEnumerable<Group> Get(GroupQuery query);
 
+    /// <summary>
+    /// Get a list of visible groups for the authenticated user matching the query.
+    /// When accessed without authentication, only public groups are returned.
+    /// See https://docs.gitlab.com/ee/api/groups.html#list-groups.
+    /// </summary>
+    /// <param name="query">The query parameters.</param>
+    /// <returns>All matching groups.</returns>
     GitLabCollectionResponse<Group> GetAsync(GroupQuery query);
 
-    Group this[int id] { get; }
-
     /// <summary>
-    /// Returns the project with the provided full path in the form Namespace/Name.
+    /// Get a page of visible groups for the authenticated user matching the query.
+    /// When accessed without authentication, only public groups are returned.
+    /// See https://docs.gitlab.com/ee/api/groups.html#list-groups.
     /// </summary>
-    /// <remarks>https://github.com/gitlabhq/gitlabhq/blob/master/doc/api/groups.md#details-of-a-group</remarks>
+    /// <param name="query">The query parameters, including the page.</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    /// <returns>A page of groups and the total number of matching groups (used to compute the last page).</returns>
+    /// <remarks>
+    /// The page will be empty after the last page is read.<para/>
+    /// The total maybe null if there are more than 10,000 items. See https://docs.gitlab.com/ee/user/gitlab_com/index.html#pagination-response-headers.
+    /// </remarks>
+    Task<(IReadOnlyCollection<Group> Page, int? Total)> PageAsync(PageQuery<GroupQuery> query, CancellationToken cancellationToken = default);
```

Comment:
Not a huge fan of exposed Tuples. It looks a bit cumbersome as it's repeated in multiple places. Although it might not happen, if we need to add a property in the future it makes a lot of place to change too. 

Could we create a simple class for it (or a record)? `PaginatedResponse<T>` or something equivalent

Reply:
Good idea. Created `PagedResponse<T>` type.